### PR TITLE
fix(raster): RS_Envelope returns axis-aligned bounding box for skewed rasters

### PR DESCRIPTION
**Pull request by Kontinuation**
## Summary

- **Fix `RS_Envelope`** to return the axis-aligned bounding box (AABB) instead of the convex hull for skewed/rotated rasters, matching PostGIS `ST_Envelope` semantics.
- **Fix `generate_test_rasters`** to give raster `i=0` non-zero scales via `i.max(1)`, so it has an invertible geotransform.
- **Add `build_noninvertible_raster()`** helper to `sedona-testing` for tests that need a raster with zero scales/skews.
- Update cascading test expectations in `rs_envelope`, `rs_geotransform`, and `rs_rastercoordinate`.

### CODE REVIEW

File: rust/sedona-raster-functions/src/rs_envelope.rs [hunk 1]
```diff
@@ -94,25 +94,37 @@ impl SedonaScalarKernel for RsEnvelope {
...
+
     write_wkb_polygon(
         out,
-        [(ulx, uly), (urx, ury), (lrx, lry), (llx, lly), (ulx, uly)].into_iter(),
+        [
+            (min_x, max_y),
+            (max_x, max_y),
+            (max_x, min_y),
+            (min_x, min_y),
+            (min_x, max_y),
```
[the remark this answers is not in the document]
Comment (Kontinuation) [reply rust/sedona-raster-functions/src/rs_envelope.rs]: Fixed coordinate order and tests

File: rust/sedona-testing/src/rasters.rs [hunk 2]
```diff
@@ -149,6 +149,39 @@ pub fn generate_tiled_rasters(
...
+            storage_type: StorageType::InDb,
+            outdb_url: None,
+            outdb_band_id: None,
+        })
+        .expect("start band");
+    builder.band_data_writer().append_value([0u8]);
+    builder.finish_band().expect("finish band");
+    builder.finish_raster().expect("finish raster");
+    builder.finish().expect("finish")
+}
```
[the remark this answers is not in the document]
Comment (paleolimbot) [reply rust/sedona-testing/src/rasters.rs:183]: All the other test data builders unwrap in place so I think this is OK.
